README.md

```diff
@@ -1,9 +1,33 @@
+
 # Minionify
-An app that allows basic control over remote devices.
 
+## Introduction
+An app that allows basic control over remote devices.
 The users would be divided into 2 groups; managers and minions.
 A manager can use the app to execute certain operations on the minions' devices, such as gathering statistics, sending notifications, determining location, opening another app etc.
-
 The minions would use the app passively; by using the app, they willingly give the managers permission to execute operations on their devices.
+Each minion can have only one manager, while managers can have multiple minions.
 
 Possible users of the app are worried parents with their children or strict bosses with their employees.
+
+
+## Modes
+The Minionify app will have two different modes that it can operate.
+- As A Minion: Minionify will run a background service that will listen for incoming requests from the managers, execute them and send back a response back to the manager.
+
+- As A Manager: Minionify will present a GUI to the user that will let them to control their minions apps.
+For example, a user can choose one of its minions and get their location, statistics and more...
+
+## Protocol
+The Minionify app will use Firebase as its server. 
+A manager will be able to send requests to a minion through Firebase (using to Firebase storage), in order to get information about/from the minion.
+
+Each minion will have a main folder that will have two folders nested in it:
```

Comment:
I think using the file infrastructure only and not the db is a bad utilization of firebase's services. the db is realtime, which means you can receive a callback when it is changed, and that's how you're supposed to use it, so i think you should use the db to notify the minion about a new request and the manager about a request's completion, and if you have to pass something bigger (file for example) using the storage with a reference in the db is a good idea. the db supports java serialization using json (a really good serialization from google, works really well with defined dataclasses) and it can be used to easily pass data-objects like `Request` or `Response` with fields like `fullFilePath` for references to the storage in case that's needed.

Reply:
Done.